frontend/JavaScript/index.js

```diff
@@ -0,0 +1,296 @@
+function sortedView(sortKey){
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  let prod_query=null;
+  let catLevel1=null;
+  let catLevel2=null;
+  let pageNumber = 1;
+  prod_query = urlParams.get('q');
+  catLevel1 = urlParams.get('cat1');
+  catLevel2 = urlParams.get('cat2');
+  urlSortKey = urlParams.get('sort')
+
+  if (prod_query!=null){
+    window.parent.location = `index.html?q=${prod_query}&page=1&sort=${sortKey}`
+  }
+
+  else if (catLevel1!=null ){
+    window.parent.location = `index.html?cat1=${catLevel1}&cat2=${catLevel2}&page=${pageNumber}&sort=${sortKey}`
+  }
+
+    else {
+      window.parent.location = `index.html?q=*&page=1&sort=${sortKey}`
+  }
+}
+
+window.onload=function reload(){
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  
+  var prod_query =urlParams.get('q');
+  if(prod_query==""){
+    var prod_query='*';
+  }
+  var catLevel1 =urlParams.get('cat1');
+  var catLevel2 =urlParams.get('cat2');
+  var pageNumber = urlParams.get('page')
+  var sortKey = urlParams.get('sort')
+  
+  if (pageNumber==null){
+    pageNumber = 1;
+  }
+
+  if (sortKey==null){
+    sortKey = '';
+  }
+
+  fetch(`http://127.0.0.1:7002/categoryTree`,{
```

Comment:
The fetch code seems to be exactly same code as used in other files , if it is the same code try to reuse it by adding it in single file and then importing it from there.

Reply:
fetch code for the category tree is now imported from the category.js and redundancy is removed.